goss-testing/scripts/check_network_interface.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+# On failures we will exit with unusual exit codes, to help make it clearer where the test failed
+
+# This test uses command pipelines, so we will set pipefail. That way the pipelines will
+# fail (i.e. exit non-0) if any of the commands in the chain fail
+set -o pipefail
+
+function err_exit
+{
+    local rc
+    rc=$1
+    shift
+    echo "ERROR: $*" 1>&2
+    echo "FAIL"
+    exit $rc
+}
+
+TESTIF=$1
```

Comment:
I don't know why, but I always like to validate the inputs to scripts. If nothing else it can detect if something upstream has gone haywire and is passing in weird stuff.

In terms of echoing the argument, the test logger I wrote will log this itself, but I like to print the arguments to scripts. This is mainly in case someone calls it manually, just because every once in a while someone uses a shell loop to run a script and end up passing in an unexpected value. So I like to show them what they called it with.

```suggestion
[ $# -eq 1 ] || err_exit 5 "Test requires exactly 1 argument but received $#: $*"
[ -n "$1" ] || err_exit 7 "Argument to test may not be blank"

TESTIF=$1

echo "Test interface: $TESTIF"
```

Reply:
Will address in a later PR